internal/external-dns/plan/conflict.go

```diff
@@ -48,15 +52,19 @@ func (s PerResource) ResolveCreate(candidates []*endpoint.Endpoint) *endpoint.En
 // ResolveUpdate is invoked when dns name is already owned by "current" endpoint
 // ResolveUpdate uses "current" record as base and updates it accordingly with new version of same resource
 // if it doesn't exist then pick min
-func (s PerResource) ResolveUpdate(current *endpoint.Endpoint, candidates []*endpoint.Endpoint) *endpoint.Endpoint {
-	currentResource := current.Labels[endpoint.ResourceLabelKey] // resource which has already acquired the DNS
+func (s PerResource) ResolveUpdate(current *endpoint.Endpoint, owner string, candidates []*endpoint.Endpoint) *endpoint.Endpoint {
+	ownedLabels := s.packer.UnpackLabels(current.Labels)[owner]
+	if ownedLabels == nil {
+		ownedLabels = current.Labels
+	}
+	currentResource := ownedLabels[endpoint.ResourceLabelKey] // resource which has already acquired the DNS
 	// TODO: sort candidates only needed because we can still have two endpoints from same resource here. We sort for consistency
 	// TODO: remove once single endpoint can have multiple targets
 	sort.SliceStable(candidates, func(i, j int) bool {
 		return s.less(candidates[i], candidates[j])
 	})
 	for _, ep := range candidates {
-		if ep.Labels[endpoint.ResourceLabelKey] == currentResource {
+		if ep.Labels[endpoint.ResourceLabelKey] == currentResource || s.packer.UnpackLabels(ep.Labels)[owner][endpoint.ResourceLabelKey] == currentResource {
```

Comment:
do we need the same nil check here as is on 57?

Reply:
If we only put endpoints from the registry here, they will never have nil labels, but they _could_ be nil here, so sure